core/worker.go

```diff
@@ -1403,9 +1611,54 @@ func (w *worker) fillTransactions(interrupt *int32, env *environment, block *typ
 
 	pendingQiTxs := w.txPool.QiPoolPending()
 
-	if len(pending) > 0 || len(pendingQiTxs) > 0 || etxs {
-		txs := types.NewTransactionsByPriceAndNonce(env.signer, pendingQiTxs, pending, env.wo.BaseFee(), true)
-		return w.commitTransactions(env, block, txs, interrupt)
+	// Convert these pendingQiTxs fees into Quai fees
+	pendingQiTxsWithQuaiFee := make([]*types.TxWithMinerFee, 0)
+	for _, tx := range pendingQiTxs {
+		// update the fee
+		qiFeeInQuai := misc.QiToQuai(primeTerminus, tx.Tx().GasPrice())
```

Comment:
This will panic (Qi tx does not have gasPrice)

Reply:
Idk how we even do this then, because unless we have a method that calculates the fee in Qi for a given Qi transaction, we cannot get the fee. I guess i make one

Reply:
Turns out we need the databse to actually look up the tx to get the price, no other way around it

Reply:
Found a solution, just have to read the MinerFee, changed it